src/adapter/vertex.rs

```diff
@@ -254,6 +256,13 @@ impl<'a> Vertex<'a> {
             _ => None,
         }
     }
+
+    pub(super) fn as_discriminant(&self) -> Option<rustdoc_types::Discriminant> {
+        match &self.kind {
+            VertexKind::Discriminant(discriminant) => Some(discriminant.clone()),
```

Comment:
If you choose to make the vertex hold a reference, then this `.clone()` would be avoided:
```suggestion
            VertexKind::Discriminant(discriminant) => Some(discriminant),
```

Reply:
See https://github.com/obi1kenobi/trustfall-rustdoc-adapter/pull/337#discussion_r1685854142.